src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/MakeTypesInternal.cs

```diff
@@ -32,7 +33,10 @@ public override void Initialize(AnalysisContext context)
             context.RegisterCompilationStartAction(context =>
             {
                 var compilation = context.Compilation;
-                if (compilation.Options.OutputKind is not (OutputKind.ConsoleApplication or OutputKind.WindowsApplication or OutputKind.WindowsRuntimeApplication))
+                SyntaxTree? firstSyntaxTree;
+                if (compilation.Options.OutputKind is not (OutputKind.ConsoleApplication or OutputKind.WindowsApplication or OutputKind.WindowsRuntimeApplication)
+                    || (firstSyntaxTree = context.Compilation.SyntaxTrees.FirstOrDefault()) is null
+                    || !context.Options.GetOutputKindsOption(Rule, firstSyntaxTree, compilation).Contains(compilation.Options.OutputKind))
```

Comment:
Thanks for making OutputKind configurable, though looks with the current logic only `OutputKind.ConsoleApplication, OutputKind.WindowsApplication or OutputKind.WindowsRuntimeApplication` are configurable, there is no way to enable the rule for `OutputKind.DynamicallyLinkedLibrary`. I think if it is giving an option to configure, all should be configurable, what you think @CollinAlpert @stephentoub?

Also please add the rule into `Configurable Rules` list in https://github.com/dotnet/roslyn-analyzers/blob/main/docs/Analyzer%20Configuration.md#analyzed-output-kinds

Reply:
I agree, it is better to make all `OutputKind`s configurable.